### PR TITLE
Allowed extension to use clipboard content to generate QR code when permission given

### PR DESCRIPTION
**Pull request by BrandonTuTwo2**
PR for this issue https://github.com/rugk/offline-qr-code/issues/282. Allows the user to create QR codes from copied text from the clipboard granted they give permission to allow the extension to read from the clipboard and turn the setting on. If the clipboard is empty defaults the current tabs URL

### CODE REVIEW

**Comment by BrandonTuTwo2** [comment]
Hi @rugk I'm ready for a review whenever your free!

File: src/options/modules/CustomOptionTriggers.js [hunk 1]
```diff
@@ -59,6 +67,9 @@ function applyQrCodeSize(optionValue) {
     }
 }
 
+
```
Comment (rugk) [inline src/options/modules/CustomOptionTriggers.js:70]: I guess remove the superfluous empty lines.

File: src/options/modules/CustomOptionTriggers.js [hunk 2]
```diff
@@ -254,6 +265,29 @@ function resetOnBeforeLoad() {
     elQrCodeSize.removeAttribute("disabled");
 }
 
+
```
Comment (rugk) [inline src/options/modules/CustomOptionTriggers.js:268]: Also the first two lines are not needed here.

File: src/options/options.html [hunk 3]
```diff
@@ -180,14 +180,29 @@ <h1 data-i18n="__MSG_titleAddonBehavior__">Addon behavior</h1>
 							<label data-i18n="__MSG_optionAutoGetSelectedText__" for="autoGetSelectedText">Automatically use text selected on website</label>
 						</div>
 					</li>
-
+
```
Comment (rugk) [inline src/options/options.html:183]: Also remove that hehe

**Review by rugk — CHANGES_REQUESTED** [review]
First review just showed some superfluous lines.

Thanks a lot!

**Comment by BrandonTuTwo2** [comment]
Hi @rugk I fixed the spacing issues and added the missing semicolons let me know if there is anything else!

**Review by rugk — APPROVED** [review]

**Comment by rugk** [comment]
Tested it and it works, except the permission prompt. When you remove the permission (and toggle the setting), you are not prompted for a new permission hmm:
grafik

Ah okay, that was just me missing the submodule, i see.

Anyway, there is a translation missing here :upside_down_face: :
grafik

**Comment by rugk** [comment]
Okay that was also partially my bad, sorry, because the lib was not really documented in a good way. Added that now/did that now: https://github.com/TinyWebEx/PermissionRequest/tree/master?tab=readme-ov-file#required-translation

And fixed the issue here.

**Comment by rugk** [comment]
BTW: Next time, try to avoid doing a pull request from the main/master branch, because you can run into problems when you have a "non-clean" main that does not follow this repo here (i.e. "upstream"). [See this article for details](https://help.github.com/en/articles/proposing-changes-to-your-work-with-pull-requests). Anyway, this is only a tip for the next time. :smiley:

**Comment by rugk** [comment]
Okay, fixed a small style issue, where the height would be too much when no permission request was shown below the option. See https://github.com/rugk/offline-qr-code/pull/337/commits/71f0932c3509cf5ffb940f449215303999dd71a0

**Comment by rugk** [comment]
Next time, FYI you can (automatically) [let issues close](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) when a PR is merged by adding some "magic" text to your PR _body_. (Manually did it now.)

**Pull request merged**